src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,152 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+    private Node<K, V>[] table = new Node[DEFAULT_CAPACITY];
+    private int size = 0;
 
     @Override
     public void put(K key, V value) {
-
+        if (threshold() == size) {
+            resize();
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        linkToBucket(table, newNode);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (size == 0) {
+            return null;
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        int position = hash % table.length;
+        position = validatePosition(position, table);
+        Node<K, V> current = new Node<>(hash, key, null, null);
+        return findNode(position, current);
```

Comment:
what we can do instead of `findNode()`? - is a `while table[position] != null` - compare keys and return value if true, and null otherwise